compliance_checker/tests/test_cf.py

```diff
@@ -1078,11 +1078,57 @@ def test_check_calendar(self):
         dataset = self.load_dataset(STATIC_FILES["bad"])
         results = self.cf.check_calendar(dataset)
         scored, out_of, messages = get_results(results)
+        # TEST CONFORMANCE 4.4.1 REQUIRED 2, 3 / 5
+        bad_month_msg = ("For nonstandard calendar on variable time, attribute "
+                         "month_lengths must be supplied as a 12-element integer array")
 
-        assert (
-            u"§4.4.1 Variable time should have a valid calendar: 'nope' is not a valid calendar"
-            in messages
-        )
+        assert bad_month_msg in messages
+
+        dataset = MockTimeSeries()
+        time = dataset.variables["time"]
+        dataset.variables["time"].calendar = "custom"
+        dataset.variables["time"].month_lengths = np.array([30.3],
+                                                           dtype=np.double)
+        results = self.cf.check_calendar(dataset)
+        scored, out_of, messages = get_results(results)
+        assert bad_month_msg in messages
+
+        dataset.variables["time"].month_lengths = np.array([31, 29, 31, 30, 31,
+                                                            30, 31, 31, 30, 31,
+                                                            30, 31],
+                                                            dtype=np.int)
+        results = self.cf.check_calendar(dataset)
+        scored, out_of, messages = get_results(results)
+        assert bad_month_msg not in messages
+
+        # TEST CONFORMANCE 4.4.1 REQUIRED 4,5/5
+        leap_month_msg = ("When attribute leap_month is supplied for variable "
+                          "time, the value must be a scalar integer between 1 "
+                          "and 12")
+        dataset.variables["time"].leap_month = np.array([0], dtype=np.uint8)
+        results = self.cf.check_calendar(dataset)
+        scored, out_of, messages = get_results(results)
+        assert leap_month_msg in messages
+
+        dataset.variables["time"].leap_month = 2
+        results = self.cf.check_calendar(dataset)
+        scored, out_of, messages = get_results(results)
+        assert leap_month_msg not in messages
+        # TEST CONFORMANCE 4.4.1 RECOMMENDED 1/2
```

Comment:
Let me know what you think of this format and `IMPLEMENTATION` for the implementation of the CF Conformance docs.  This portion of the code might be a later candidate for BDD libraries for the specifications outlined by the CF Conformance docs.

Reply:
Sorry I'm late but I kind like it.
(I was waiting until the end of the code sprint to get to these reviews, might as well as use the code sprint time to do them.)